docs/reference/sql/endpoints/version-compat.asciidoc

```diff
@@ -0,0 +1,17 @@
+[testenv="platinum"]
+[discrete]
+= Version compatibility
+
+You will need to choose a driver version that will be supported by your {es} server.
+For server versions prior to v7.7.0, driver's version will have to exactly match that of server's. For server versions
+on or past v7.7.0, driver's version can to be as recent as server's or earlier, but only up to one major version
+older. The driver can't be on a version more recent than server's.
+
+Examples:
+
+- a v7.6.1 server will only work with a v7.6.1 driver;
+- a v8.1.2 server will work with any driver on v8.1.2 or earlier 8.x version, as well as any 7.x version on or
+newer than v7.7.0;
+- a v9.2.4 server will work with any driver on v9.2.4 or earlier 9.x version, as well as any 8.x version, but won't
+work with any 7.x or earlier driver.
```

Comment:
Thanks, @jrodewig. There are some minor changes needed on the ES versions column and an extra clarification added to the driver's version column (compatibility on current major line). The page with the table looks better, but misses to make explicit one thing: the drivers can't be newer than the server. This deployment set-up happened, caused user confusion and actually promoted for this PR to be raised; it's somehow also naturally occurring, since the servers tend to be upgraded slower and users tend to grab latest drivers.

I could add it to the driver's table, but it's not a great fit, since that details what is allowed (not what isn't). Wondering if you might have a suggestion on how to improve accommodate this, maybe add the note after the table?

Table explanations now:
<img width="1227" alt="Screen Shot 2021-03-16 at 20 02 42" src="https://user-images.githubusercontent.com/1562801/111367169-be3eca80-8694-11eb-8bbf-1739c4cda415.png">

Reply:
Thanks for catching my omissions @bpintea. I updated by suggestion to include an admon before the table. I also updated the bullet for the current major version.